sklearn/linear_model/tests/test_sgd.py

```diff
@@ -467,6 +468,26 @@ def test_set_coef_multiclass(self):
         # Provided intercept_ does match dataset.
         clf = self.factory().fit(X2, Y2, intercept_init=np.zeros((3,)))
 
+    def test_sgd_proba_access(self):
+        # Test for issue #10938
+
+        # Checks if predict_proba and predict_log_proba
+        # is accessible for refrencing before fitting
+        # the SGD classifier
+        clf = SGDClassifier()
+        assert not(hasattr(clf, "predict_proba"))
```

Comment:
Maybe check this for all the losses that do not support predict_proba, you can use `SGD.loss_functions` to get all the possible losses I think.

Slight improvement would be to also check the error message:
```py
with pytest.raises(AttributeError, 'probability estimates are not available for loss={!r}'.format(loss)
```